src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -181,28 +194,28 @@ module MOM_ice_shelf
 
   logical :: debug                !< If true, write verbose checksums for debugging purposes
                                   !! and use reproducible sums
+  integer :: id_clock_shelf=-1 !< CPU Clock for the ice shelf code
+  integer :: id_clock_pass=-1 !< CPU Clock for group pass calls
 end type ice_shelf_CS
 
-integer :: id_clock_shelf !< CPU Clock for the ice shelf code
-integer :: id_clock_pass !< CPU Clock for group pass calls
+
 
 contains
 
 !> Calculates fluxes between the ocean and ice-shelf using the three-equations
 !! formulation (optional to use just two equations).
 !! See \ref section_ICE_SHELF_equations
-subroutine shelf_calc_flux(sfc_state, fluxes, Time, time_step, CS, forces)
-  type(surface),         intent(inout) :: sfc_state !< A structure containing fields that
+subroutine shelf_calc_flux(sfc_state_in, fluxes_in, Time, time_step, CS)
+  type(surface), target,         intent(inout) :: sfc_state_in !< A structure containing fields that
                                                 !! describe the surface state of the ocean.  The
                                                 !! intent is only inout to allow for halo updates.
-  type(forcing),         intent(inout) :: fluxes !< structure containing pointers to any possible
+  type(forcing),         pointer    :: fluxes_in !< structure containing pointers to any possible
```

Comment:
This (and other `*_in`) forcing inputs should probably be declared as `target` rather than `pointer`.